### PR TITLE
[lz4] depends on xxhash

### LINKED ISSUE #28747: [xxhash] file conflict with lz4
From CI
```
2023-01-05T11:17:07.5775394Z Building xxhash[core]:arm-uwp...
2023-01-05T11:17:07.6131083Z -- Downloading https://github.com/Cyan4973/xxHash/archive/v0.8.1.tar.gz -> Cyan4973-xxHash-v0.8.1.tar.gz...
2023-01-05T11:17:07.7021363Z -- Extracting source D:/downloads/Cyan4973-xxHash-v0.8.1.tar.gz
2023-01-05T11:17:07.8340030Z -- Using source at D:/buildtrees/xxhash/src/v0.8.1-c171284619.clean
2023-01-05T11:17:07.8453021Z -- Found external ninja('1.11.0').
2023-01-05T11:17:07.8491537Z -- Configuring arm-uwp
2023-01-05T11:17:08.4377865Z -- Building arm-uwp-dbg
2023-01-05T11:17:08.6400756Z -- Building arm-uwp-rel
2023-01-05T11:17:09.1332341Z -- Fixing pkgconfig file: D:/packages/xxhash_arm-uwp/lib/pkgconfig/libxxhash.pc
2023-01-05T11:17:09.1402322Z -- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
2023-01-05T11:17:09.1431793Z -- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
2023-01-05T11:17:09.1496354Z -- Using msys root at D:/downloads/tools/msys2/9a1ec3f33446b195
2023-01-05T11:17:09.1611662Z -- Fixing pkgconfig file: D:/packages/xxhash_arm-uwp/debug/lib/pkgconfig/libxxhash.pc
2023-01-05T11:17:09.1732802Z -- Installing: D:/packages/xxhash_arm-uwp/share/xxhash/copyright
2023-01-05T11:17:09.1785397Z -- Performing post-build validation
2023-01-05T11:17:09.4780531Z Uploaded binaries to 1 HTTP remotes.
2023-01-05T11:17:09.5759382Z error: The following files are already installed in D:/installed/arm-uwp and are in conflict with xxhash:arm-uwp
2023-01-05T11:17:09.5765084Z Installed by lz4:arm-uwp    
2023-01-05T11:17:09.5771017Z include/xxhash.h
2023-01-05T11:17:09.5773701Z 
2023-01-05T11:17:09.5886020Z Elapsed time to handle xxhash:arm-uwp: 2.013 s
```

### PR DESCRIPTION
**Pull request by autoantwort**
Fixes #28747

### CODE REVIEW

File: versions/baseline.json [hunk 1]
```diff
@@ -6060,6 +6060,10 @@
       "baseline": "2021-02-21",
       "port-version": 0
     },
+    "ptc-print": {
+      "baseline": "1.4.0",
+      "port-version": 0
+    },
```
Comment (autoantwort) [inline versions/baseline.json:6066]: arrgh..

File: ports/lz4/CMakeLists.txt [hunk 2]
```diff
@@ -19,9 +19,11 @@ add_library(lz4
     lib/lz4.c
     lib/lz4frame.c
     lib/lz4hc.c
-    lib/xxhash.c
 )
 
+find_package(xxHash CONFIG REQUIRED)
```
Comment (Neumann-A) [inline ports/lz4/CMakeLists.txt:24]: does this need a correction of the pc file or is there no pkg-config file

Comment (autoantwort) [reply ports/lz4/CMakeLists.txt:24]: Nice catch. I have added `-lxxhash`

**Comment by Neumann-A** [comment]
You forgot to add the `find_dependency` call in the `lz4-config.cmake`. 
You also might want to add a `#pragma comment(lib, <xxhashlib>)` to one of the main headers to simply autolink. Don't know if gcc has something similar.

**Comment by autoantwort** [comment]
> You forgot to add the `find_dependency` call in the `lz4-config.cmake`.

Can I also do that with 
```cmake
install(EXPORT lz4Config
  FILE lz4-config.cmake
  NAMESPACE lz4::
  DESTINATION "${INSTALL_CMAKE_DIR}"
)
```

**Comment by Neumann-A** [comment]
> Can I also do that with

only by reading that file from the portfile and adding the lines at the top.

**Comment by dg0yt** [comment]
> You also might want to add a `#pragma comment(lib, <xxhashlib>)` to one of the main headers to simply autolink. Don't know if gcc has something similar.

This exists only in MSVC. 

AFAICT lz4 would use the vendored xxhash with a specific name prefix (at least if not using vcpkg's vendored build system). As immediate fix, it would have been enough to not install this header. As another step, consider upstream's optional meson build.

**Comment by Neumann-A** [comment]
> As another step, consider upstream's optional meson build.

Probably easier and gets rid of the CMakeLists.txt vendored by vcpkg.

**Comment by dg0yt** [comment]
@vicroms We need this one merged ASAP. Baseline regression.

**Review by FrankXie05 — APPROVED** [review]

**Comment by FrankXie05** [comment]
Temporarily merging it to fix issues with baseline.

**Pull request merged**